// @pulumi/cloudflare/accessMutualTlsHostnameSettings.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessMutualTlsHostnameSettings = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * @deprecated cloudflare.index/accessmutualtlshostnamesettings.AccessMutualTlsHostnameSettings has been deprecated in favor of cloudflare.index/zerotrustaccessmtlshostnamesettings.ZeroTrustAccessMtlsHostnameSettings
 */
class AccessMutualTlsHostnameSettings extends pulumi.CustomResource {
    /**
     * Get an existing AccessMutualTlsHostnameSettings resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        pulumi.log.warn("AccessMutualTlsHostnameSettings is deprecated: cloudflare.index/accessmutualtlshostnamesettings.AccessMutualTlsHostnameSettings has been deprecated in favor of cloudflare.index/zerotrustaccessmtlshostnamesettings.ZeroTrustAccessMtlsHostnameSettings");
        return new AccessMutualTlsHostnameSettings(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AccessMutualTlsHostnameSettings.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AccessMutualTlsHostnameSettings.__pulumiType;
    }
    /** @deprecated cloudflare.index/accessmutualtlshostnamesettings.AccessMutualTlsHostnameSettings has been deprecated in favor of cloudflare.index/zerotrustaccessmtlshostnamesettings.ZeroTrustAccessMtlsHostnameSettings */
    constructor(name, argsOrState, opts) {
        pulumi.log.warn("AccessMutualTlsHostnameSettings is deprecated: cloudflare.index/accessmutualtlshostnamesettings.AccessMutualTlsHostnameSettings has been deprecated in favor of cloudflare.index/zerotrustaccessmtlshostnamesettings.ZeroTrustAccessMtlsHostnameSettings");
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["chinaNetwork"] = state ? state.chinaNetwork : undefined;
            resourceInputs["clientCertificateForwarding"] = state ? state.clientCertificateForwarding : undefined;
            resourceInputs["hostname"] = state ? state.hostname : undefined;
            resourceInputs["settings"] = state ? state.settings : undefined;
            resourceInputs["zoneId"] = state ? state.zoneId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.settings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'settings'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["settings"] = args ? args.settings : undefined;
            resourceInputs["zoneId"] = args ? args.zoneId : undefined;
            resourceInputs["chinaNetwork"] = undefined /*out*/;
            resourceInputs["clientCertificateForwarding"] = undefined /*out*/;
            resourceInputs["hostname"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "cloudflare:index/accessMutualTlsHostnameSettings:AccessMutualTlsHostnameSettings" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AccessMutualTlsHostnameSettings.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AccessMutualTlsHostnameSettings = AccessMutualTlsHostnameSettings;
/** @internal */
AccessMutualTlsHostnameSettings.__pulumiType = 'cloudflare:index/accessMutualTlsHostnameSettings:AccessMutualTlsHostnameSettings';
