// @pulumi/cloudflare/addressMap.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddressMap = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleAddressMap = new cloudflare.AddressMap("example_address_map", {
 *     accountId: "258def64c72dae45f3e4c8516e2111f2",
 *     description: "My Ecommerce zones",
 *     enabled: true,
 *     ips: ["192.0.2.1"],
 *     memberships: [{
 *         identifier: "023e105f4ecef8ad9ca31a8372d0c353",
 *         kind: "zone",
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/addressMap:AddressMap example '<account_id>/<address_map_id>'
 * ```
 */
class AddressMap extends pulumi.CustomResource {
    /**
     * Get an existing AddressMap resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AddressMap(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AddressMap.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AddressMap.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["canDelete"] = state ? state.canDelete : undefined;
            resourceInputs["canModifyIps"] = state ? state.canModifyIps : undefined;
            resourceInputs["createdAt"] = state ? state.createdAt : undefined;
            resourceInputs["defaultSni"] = state ? state.defaultSni : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["enabled"] = state ? state.enabled : undefined;
            resourceInputs["ips"] = state ? state.ips : undefined;
            resourceInputs["memberships"] = state ? state.memberships : undefined;
            resourceInputs["modifiedAt"] = state ? state.modifiedAt : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["defaultSni"] = args ? args.defaultSni : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["ips"] = args ? args.ips : undefined;
            resourceInputs["memberships"] = args ? args.memberships : undefined;
            resourceInputs["canDelete"] = undefined /*out*/;
            resourceInputs["canModifyIps"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["modifiedAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AddressMap.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AddressMap = AddressMap;
/** @internal */
AddressMap.__pulumiType = 'cloudflare:index/addressMap:AddressMap';
