// @pulumi/cloudflare/apiShieldSchema.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiShieldSchema = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleApiShieldSchema = new cloudflare.ApiShieldSchema("example_api_shield_schema", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     file: null,
 *     kind: "openapi_v3",
 *     name: "petstore schema",
 *     validationEnabled: "true",
 * });
 * ```
 */
class ApiShieldSchema extends pulumi.CustomResource {
    /**
     * Get an existing ApiShieldSchema resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ApiShieldSchema(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ApiShieldSchema.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ApiShieldSchema.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createdAt"] = state ? state.createdAt : undefined;
            resourceInputs["file"] = state ? state.file : undefined;
            resourceInputs["kind"] = state ? state.kind : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["schema"] = state ? state.schema : undefined;
            resourceInputs["schemaId"] = state ? state.schemaId : undefined;
            resourceInputs["source"] = state ? state.source : undefined;
            resourceInputs["uploadDetails"] = state ? state.uploadDetails : undefined;
            resourceInputs["validationEnabled"] = state ? state.validationEnabled : undefined;
            resourceInputs["zoneId"] = state ? state.zoneId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.file === undefined) && !opts.urn) {
                throw new Error("Missing required property 'file'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.zoneId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["file"] = args ? args.file : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["schemaId"] = args ? args.schemaId : undefined;
            resourceInputs["validationEnabled"] = args ? args.validationEnabled : undefined;
            resourceInputs["zoneId"] = args ? args.zoneId : undefined;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["schema"] = undefined /*out*/;
            resourceInputs["source"] = undefined /*out*/;
            resourceInputs["uploadDetails"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ApiShieldSchema.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ApiShieldSchema = ApiShieldSchema;
/** @internal */
ApiShieldSchema.__pulumiType = 'cloudflare:index/apiShieldSchema:ApiShieldSchema';
