// @pulumi/cloudflare/dnsFirewall.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DnsFirewall = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/dnsFirewall:DnsFirewall example '<account_id>/<dns_firewall_id>'
 * ```
 */
class DnsFirewall extends pulumi.CustomResource {
    /**
     * Get an existing DnsFirewall resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DnsFirewall(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DnsFirewall.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DnsFirewall.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["attackMitigation"] = state ? state.attackMitigation : undefined;
            resourceInputs["deprecateAnyRequests"] = state ? state.deprecateAnyRequests : undefined;
            resourceInputs["dnsFirewallIps"] = state ? state.dnsFirewallIps : undefined;
            resourceInputs["ecsFallback"] = state ? state.ecsFallback : undefined;
            resourceInputs["maximumCacheTtl"] = state ? state.maximumCacheTtl : undefined;
            resourceInputs["minimumCacheTtl"] = state ? state.minimumCacheTtl : undefined;
            resourceInputs["modifiedOn"] = state ? state.modifiedOn : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["negativeCacheTtl"] = state ? state.negativeCacheTtl : undefined;
            resourceInputs["ratelimit"] = state ? state.ratelimit : undefined;
            resourceInputs["retries"] = state ? state.retries : undefined;
            resourceInputs["upstreamIps"] = state ? state.upstreamIps : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.upstreamIps === undefined) && !opts.urn) {
                throw new Error("Missing required property 'upstreamIps'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["attackMitigation"] = args ? args.attackMitigation : undefined;
            resourceInputs["deprecateAnyRequests"] = args ? args.deprecateAnyRequests : undefined;
            resourceInputs["ecsFallback"] = args ? args.ecsFallback : undefined;
            resourceInputs["maximumCacheTtl"] = args ? args.maximumCacheTtl : undefined;
            resourceInputs["minimumCacheTtl"] = args ? args.minimumCacheTtl : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["negativeCacheTtl"] = args ? args.negativeCacheTtl : undefined;
            resourceInputs["ratelimit"] = args ? args.ratelimit : undefined;
            resourceInputs["retries"] = args ? args.retries : undefined;
            resourceInputs["upstreamIps"] = args ? args.upstreamIps : undefined;
            resourceInputs["dnsFirewallIps"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DnsFirewall.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DnsFirewall = DnsFirewall;
/** @internal */
DnsFirewall.__pulumiType = 'cloudflare:index/dnsFirewall:DnsFirewall';
