// @pulumi/cloudflare/emailSecurityTrustedDomains.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailSecurityTrustedDomains = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleEmailSecurityTrustedDomains = new cloudflare.EmailSecurityTrustedDomains("example_email_security_trusted_domains", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     isRecent: true,
 *     isRegex: true,
 *     isSimilarity: true,
 *     pattern: "x",
 *     comments: "comments",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/emailSecurityTrustedDomains:EmailSecurityTrustedDomains example '<account_id>/<trusted_domain_id>'
 * ```
 */
class EmailSecurityTrustedDomains extends pulumi.CustomResource {
    /**
     * Get an existing EmailSecurityTrustedDomains resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EmailSecurityTrustedDomains(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EmailSecurityTrustedDomains.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EmailSecurityTrustedDomains.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["bodies"] = state ? state.bodies : undefined;
            resourceInputs["comments"] = state ? state.comments : undefined;
            resourceInputs["createdAt"] = state ? state.createdAt : undefined;
            resourceInputs["isRecent"] = state ? state.isRecent : undefined;
            resourceInputs["isRegex"] = state ? state.isRegex : undefined;
            resourceInputs["isSimilarity"] = state ? state.isSimilarity : undefined;
            resourceInputs["lastModified"] = state ? state.lastModified : undefined;
            resourceInputs["pattern"] = state ? state.pattern : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["bodies"] = args ? args.bodies : undefined;
            resourceInputs["comments"] = args ? args.comments : undefined;
            resourceInputs["isRecent"] = args ? args.isRecent : undefined;
            resourceInputs["isRegex"] = args ? args.isRegex : undefined;
            resourceInputs["isSimilarity"] = args ? args.isSimilarity : undefined;
            resourceInputs["pattern"] = args ? args.pattern : undefined;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["lastModified"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EmailSecurityTrustedDomains.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EmailSecurityTrustedDomains = EmailSecurityTrustedDomains;
/** @internal */
EmailSecurityTrustedDomains.__pulumiType = 'cloudflare:index/emailSecurityTrustedDomains:EmailSecurityTrustedDomains';
