// @pulumi/cloudflare/imageVariant.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageVariant = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/imageVariant:ImageVariant example '<account_id>/<variant_id>'
 * ```
 */
class ImageVariant extends pulumi.CustomResource {
    /**
     * Get an existing ImageVariant resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ImageVariant(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ImageVariant.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ImageVariant.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["neverRequireSignedUrls"] = state ? state.neverRequireSignedUrls : undefined;
            resourceInputs["options"] = state ? state.options : undefined;
            resourceInputs["variant"] = state ? state.variant : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.options === undefined) && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["neverRequireSignedUrls"] = args ? args.neverRequireSignedUrls : undefined;
            resourceInputs["options"] = args ? args.options : undefined;
            resourceInputs["variant"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ImageVariant.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ImageVariant = ImageVariant;
/** @internal */
ImageVariant.__pulumiType = 'cloudflare:index/imageVariant:ImageVariant';
