// @pulumi/cloudflare/magicTransitSiteWan.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MagicTransitSiteWan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/magicTransitSiteWan:MagicTransitSiteWan example '<account_id>/<site_id>/<wan_id>'
 * ```
 */
class MagicTransitSiteWan extends pulumi.CustomResource {
    /**
     * Get an existing MagicTransitSiteWan resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MagicTransitSiteWan(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MagicTransitSiteWan.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MagicTransitSiteWan.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["healthCheckRate"] = state ? state.healthCheckRate : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["physport"] = state ? state.physport : undefined;
            resourceInputs["priority"] = state ? state.priority : undefined;
            resourceInputs["siteId"] = state ? state.siteId : undefined;
            resourceInputs["staticAddressing"] = state ? state.staticAddressing : undefined;
            resourceInputs["vlanTag"] = state ? state.vlanTag : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.physport === undefined) && !opts.urn) {
                throw new Error("Missing required property 'physport'");
            }
            if ((!args || args.siteId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'siteId'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["physport"] = args ? args.physport : undefined;
            resourceInputs["priority"] = args ? args.priority : undefined;
            resourceInputs["siteId"] = args ? args.siteId : undefined;
            resourceInputs["staticAddressing"] = args ? args.staticAddressing : undefined;
            resourceInputs["vlanTag"] = args ? args.vlanTag : undefined;
            resourceInputs["healthCheckRate"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MagicTransitSiteWan.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MagicTransitSiteWan = MagicTransitSiteWan;
/** @internal */
MagicTransitSiteWan.__pulumiType = 'cloudflare:index/magicTransitSiteWan:MagicTransitSiteWan';
