// @pulumi/cloudflare/queue.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Queue = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleQueue = new cloudflare.Queue("example_queue", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     queueName: "example-queue",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/queue:Queue example '<account_id>/<queue_id>'
 * ```
 */
class Queue extends pulumi.CustomResource {
    /**
     * Get an existing Queue resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Queue(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Queue.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Queue.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["consumers"] = state ? state.consumers : undefined;
            resourceInputs["consumersTotalCount"] = state ? state.consumersTotalCount : undefined;
            resourceInputs["createdOn"] = state ? state.createdOn : undefined;
            resourceInputs["modifiedOn"] = state ? state.modifiedOn : undefined;
            resourceInputs["producers"] = state ? state.producers : undefined;
            resourceInputs["producersTotalCount"] = state ? state.producersTotalCount : undefined;
            resourceInputs["queueId"] = state ? state.queueId : undefined;
            resourceInputs["queueName"] = state ? state.queueName : undefined;
            resourceInputs["settings"] = state ? state.settings : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.queueName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queueName'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["queueName"] = args ? args.queueName : undefined;
            resourceInputs["settings"] = args ? args.settings : undefined;
            resourceInputs["consumers"] = undefined /*out*/;
            resourceInputs["consumersTotalCount"] = undefined /*out*/;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
            resourceInputs["producers"] = undefined /*out*/;
            resourceInputs["producersTotalCount"] = undefined /*out*/;
            resourceInputs["queueId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Queue.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Queue = Queue;
/** @internal */
Queue.__pulumiType = 'cloudflare:index/queue:Queue';
