// @pulumi/cloudflare/queueConsumer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueueConsumer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 */
class QueueConsumer extends pulumi.CustomResource {
    /**
     * Get an existing QueueConsumer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new QueueConsumer(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of QueueConsumer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === QueueConsumer.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["consumerId"] = state ? state.consumerId : undefined;
            resourceInputs["createdOn"] = state ? state.createdOn : undefined;
            resourceInputs["deadLetterQueue"] = state ? state.deadLetterQueue : undefined;
            resourceInputs["queueId"] = state ? state.queueId : undefined;
            resourceInputs["script"] = state ? state.script : undefined;
            resourceInputs["scriptName"] = state ? state.scriptName : undefined;
            resourceInputs["settings"] = state ? state.settings : undefined;
            resourceInputs["type"] = state ? state.type : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.queueId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queueId'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["consumerId"] = args ? args.consumerId : undefined;
            resourceInputs["deadLetterQueue"] = args ? args.deadLetterQueue : undefined;
            resourceInputs["queueId"] = args ? args.queueId : undefined;
            resourceInputs["scriptName"] = args ? args.scriptName : undefined;
            resourceInputs["settings"] = args ? args.settings : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["script"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(QueueConsumer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.QueueConsumer = QueueConsumer;
/** @internal */
QueueConsumer.__pulumiType = 'cloudflare:index/queueConsumer:QueueConsumer';
