// @pulumi/cloudflare/registrarDomain.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegistrarDomain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleRegistrarDomain = new cloudflare.RegistrarDomain("example_registrar_domain", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     domainName: "cloudflare.com",
 *     autoRenew: true,
 *     locked: false,
 *     privacy: true,
 * });
 * ```
 */
class RegistrarDomain extends pulumi.CustomResource {
    /**
     * Get an existing RegistrarDomain resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RegistrarDomain(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RegistrarDomain.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegistrarDomain.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["autoRenew"] = state ? state.autoRenew : undefined;
            resourceInputs["domainName"] = state ? state.domainName : undefined;
            resourceInputs["locked"] = state ? state.locked : undefined;
            resourceInputs["privacy"] = state ? state.privacy : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.domainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["autoRenew"] = args ? args.autoRenew : undefined;
            resourceInputs["domainName"] = args ? args.domainName : undefined;
            resourceInputs["locked"] = args ? args.locked : undefined;
            resourceInputs["privacy"] = args ? args.privacy : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RegistrarDomain.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RegistrarDomain = RegistrarDomain;
/** @internal */
RegistrarDomain.__pulumiType = 'cloudflare:index/registrarDomain:RegistrarDomain';
