// @pulumi/cloudflare/streamAudioTrack.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamAudioTrack = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleStreamAudioTrack = new cloudflare.StreamAudioTrack("example_stream_audio_track", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     identifier: "ea95132c15732412d22c1476fa83f27a",
 *     audioIdentifier: "ea95132c15732412d22c1476fa83f27a",
 *     "default": true,
 *     label: "director commentary",
 * });
 * ```
 */
class StreamAudioTrack extends pulumi.CustomResource {
    /**
     * Get an existing StreamAudioTrack resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new StreamAudioTrack(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StreamAudioTrack.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StreamAudioTrack.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["audioIdentifier"] = state ? state.audioIdentifier : undefined;
            resourceInputs["default"] = state ? state.default : undefined;
            resourceInputs["identifier"] = state ? state.identifier : undefined;
            resourceInputs["label"] = state ? state.label : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
            resourceInputs["uid"] = state ? state.uid : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.identifier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'identifier'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["audioIdentifier"] = args ? args.audioIdentifier : undefined;
            resourceInputs["default"] = args ? args.default : undefined;
            resourceInputs["identifier"] = args ? args.identifier : undefined;
            resourceInputs["label"] = args ? args.label : undefined;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(StreamAudioTrack.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StreamAudioTrack = StreamAudioTrack;
/** @internal */
StreamAudioTrack.__pulumiType = 'cloudflare:index/streamAudioTrack:StreamAudioTrack';
