// @pulumi/cloudflare/streamKey.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamKey = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleStreamKey = new cloudflare.StreamKey("example_stream_key", {accountId: "023e105f4ecef8ad9ca31a8372d0c353"});
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/streamKey:StreamKey example '<account_id>'
 * ```
 */
class StreamKey extends pulumi.CustomResource {
    /**
     * Get an existing StreamKey resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new StreamKey(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StreamKey.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StreamKey.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["created"] = state ? state.created : undefined;
            resourceInputs["jwk"] = state ? state.jwk : undefined;
            resourceInputs["pem"] = state ? state.pem : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["jwk"] = undefined /*out*/;
            resourceInputs["pem"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["jwk", "pem"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(StreamKey.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StreamKey = StreamKey;
/** @internal */
StreamKey.__pulumiType = 'cloudflare:index/streamKey:StreamKey';
