// @pulumi/cloudflare/streamCaptionLanguage.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamCaptionLanguage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleStreamCaptionLanguage = new cloudflare.StreamCaptionLanguage("example_stream_caption_language", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     identifier: "ea95132c15732412d22c1476fa83f27a",
 *     language: "tr",
 * });
 * ```
 */
class StreamCaptionLanguage extends pulumi.CustomResource {
    /**
     * Get an existing StreamCaptionLanguage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new StreamCaptionLanguage(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StreamCaptionLanguage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StreamCaptionLanguage.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["file"] = state ? state.file : undefined;
            resourceInputs["generated"] = state ? state.generated : undefined;
            resourceInputs["identifier"] = state ? state.identifier : undefined;
            resourceInputs["label"] = state ? state.label : undefined;
            resourceInputs["language"] = state ? state.language : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.identifier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'identifier'");
            }
            if ((!args || args.language === undefined) && !opts.urn) {
                throw new Error("Missing required property 'language'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["file"] = args ? args.file : undefined;
            resourceInputs["identifier"] = args ? args.identifier : undefined;
            resourceInputs["language"] = args ? args.language : undefined;
            resourceInputs["generated"] = undefined /*out*/;
            resourceInputs["label"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(StreamCaptionLanguage.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StreamCaptionLanguage = StreamCaptionLanguage;
/** @internal */
StreamCaptionLanguage.__pulumiType = 'cloudflare:index/streamCaptionLanguage:StreamCaptionLanguage';
