// @pulumi/cloudflare/webAnalyticsSite.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAnalyticsSite = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleWebAnalyticsSite = new cloudflare.WebAnalyticsSite("example_web_analytics_site", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     autoInstall: true,
 *     host: "example.com",
 *     zoneTag: "023e105f4ecef8ad9ca31a8372d0c353",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/webAnalyticsSite:WebAnalyticsSite example '<account_id>/<site_id>'
 * ```
 */
class WebAnalyticsSite extends pulumi.CustomResource {
    /**
     * Get an existing WebAnalyticsSite resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new WebAnalyticsSite(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAnalyticsSite.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAnalyticsSite.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["autoInstall"] = state ? state.autoInstall : undefined;
            resourceInputs["created"] = state ? state.created : undefined;
            resourceInputs["enabled"] = state ? state.enabled : undefined;
            resourceInputs["host"] = state ? state.host : undefined;
            resourceInputs["lite"] = state ? state.lite : undefined;
            resourceInputs["rules"] = state ? state.rules : undefined;
            resourceInputs["ruleset"] = state ? state.ruleset : undefined;
            resourceInputs["siteTag"] = state ? state.siteTag : undefined;
            resourceInputs["siteToken"] = state ? state.siteToken : undefined;
            resourceInputs["snippet"] = state ? state.snippet : undefined;
            resourceInputs["zoneTag"] = state ? state.zoneTag : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["autoInstall"] = args ? args.autoInstall : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["host"] = args ? args.host : undefined;
            resourceInputs["lite"] = args ? args.lite : undefined;
            resourceInputs["zoneTag"] = args ? args.zoneTag : undefined;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["rules"] = undefined /*out*/;
            resourceInputs["ruleset"] = undefined /*out*/;
            resourceInputs["siteTag"] = undefined /*out*/;
            resourceInputs["siteToken"] = undefined /*out*/;
            resourceInputs["snippet"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(WebAnalyticsSite.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAnalyticsSite = WebAnalyticsSite;
/** @internal */
WebAnalyticsSite.__pulumiType = 'cloudflare:index/webAnalyticsSite:WebAnalyticsSite';
