// @pulumi/cloudflare/webAnalyticsRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAnalyticsRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleWebAnalyticsRule = new cloudflare.WebAnalyticsRule("example_web_analytics_rule", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     rulesetId: "f174e90a-fafe-4643-bbbc-4a0ed4fc8415",
 *     host: "example.com",
 *     inclusive: true,
 *     isPaused: false,
 *     paths: ["*"],
 * });
 * ```
 */
class WebAnalyticsRule extends pulumi.CustomResource {
    /**
     * Get an existing WebAnalyticsRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new WebAnalyticsRule(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAnalyticsRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAnalyticsRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["created"] = state ? state.created : undefined;
            resourceInputs["host"] = state ? state.host : undefined;
            resourceInputs["inclusive"] = state ? state.inclusive : undefined;
            resourceInputs["isPaused"] = state ? state.isPaused : undefined;
            resourceInputs["paths"] = state ? state.paths : undefined;
            resourceInputs["priority"] = state ? state.priority : undefined;
            resourceInputs["rulesetId"] = state ? state.rulesetId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.rulesetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rulesetId'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["host"] = args ? args.host : undefined;
            resourceInputs["inclusive"] = args ? args.inclusive : undefined;
            resourceInputs["isPaused"] = args ? args.isPaused : undefined;
            resourceInputs["paths"] = args ? args.paths : undefined;
            resourceInputs["rulesetId"] = args ? args.rulesetId : undefined;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["priority"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(WebAnalyticsRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAnalyticsRule = WebAnalyticsRule;
/** @internal */
WebAnalyticsRule.__pulumiType = 'cloudflare:index/webAnalyticsRule:WebAnalyticsRule';
