// @pulumi/cloudflare/zeroTrustAccessPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZeroTrustAccessPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/zeroTrustAccessPolicy:ZeroTrustAccessPolicy example '<account_id>/<policy_id>'
 * ```
 */
class ZeroTrustAccessPolicy extends pulumi.CustomResource {
    /**
     * Get an existing ZeroTrustAccessPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ZeroTrustAccessPolicy(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ZeroTrustAccessPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ZeroTrustAccessPolicy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["approvalGroups"] = state ? state.approvalGroups : undefined;
            resourceInputs["approvalRequired"] = state ? state.approvalRequired : undefined;
            resourceInputs["decision"] = state ? state.decision : undefined;
            resourceInputs["excludes"] = state ? state.excludes : undefined;
            resourceInputs["includes"] = state ? state.includes : undefined;
            resourceInputs["isolationRequired"] = state ? state.isolationRequired : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["purposeJustificationPrompt"] = state ? state.purposeJustificationPrompt : undefined;
            resourceInputs["purposeJustificationRequired"] = state ? state.purposeJustificationRequired : undefined;
            resourceInputs["requires"] = state ? state.requires : undefined;
            resourceInputs["sessionDuration"] = state ? state.sessionDuration : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.decision === undefined) && !opts.urn) {
                throw new Error("Missing required property 'decision'");
            }
            if ((!args || args.includes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'includes'");
            }
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["approvalGroups"] = args ? args.approvalGroups : undefined;
            resourceInputs["approvalRequired"] = args ? args.approvalRequired : undefined;
            resourceInputs["decision"] = args ? args.decision : undefined;
            resourceInputs["excludes"] = args ? args.excludes : undefined;
            resourceInputs["includes"] = args ? args.includes : undefined;
            resourceInputs["isolationRequired"] = args ? args.isolationRequired : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["purposeJustificationPrompt"] = args ? args.purposeJustificationPrompt : undefined;
            resourceInputs["purposeJustificationRequired"] = args ? args.purposeJustificationRequired : undefined;
            resourceInputs["requires"] = args ? args.requires : undefined;
            resourceInputs["sessionDuration"] = args ? args.sessionDuration : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "cloudflare:index/accessPolicy:AccessPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ZeroTrustAccessPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ZeroTrustAccessPolicy = ZeroTrustAccessPolicy;
/** @internal */
ZeroTrustAccessPolicy.__pulumiType = 'cloudflare:index/zeroTrustAccessPolicy:ZeroTrustAccessPolicy';
