// @aws-sdk/client-route-53/dist-cjs/Route53.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Route53 = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const ActivateKeySigningKeyCommand_1 = require("./commands/ActivateKeySigningKeyCommand");
const AssociateVPCWithHostedZoneCommand_1 = require("./commands/AssociateVPCWithHostedZoneCommand");
const ChangeCidrCollectionCommand_1 = require("./commands/ChangeCidrCollectionCommand");
const ChangeResourceRecordSetsCommand_1 = require("./commands/ChangeResourceRecordSetsCommand");
const ChangeTagsForResourceCommand_1 = require("./commands/ChangeTagsForResourceCommand");
const CreateCidrCollectionCommand_1 = require("./commands/CreateCidrCollectionCommand");
const CreateHealthCheckCommand_1 = require("./commands/CreateHealthCheckCommand");
const CreateHostedZoneCommand_1 = require("./commands/CreateHostedZoneCommand");
const CreateKeySigningKeyCommand_1 = require("./commands/CreateKeySigningKeyCommand");
const CreateQueryLoggingConfigCommand_1 = require("./commands/CreateQueryLoggingConfigCommand");
const CreateReusableDelegationSetCommand_1 = require("./commands/CreateReusableDelegationSetCommand");
const CreateTrafficPolicyCommand_1 = require("./commands/CreateTrafficPolicyCommand");
const CreateTrafficPolicyInstanceCommand_1 = require("./commands/CreateTrafficPolicyInstanceCommand");
const CreateTrafficPolicyVersionCommand_1 = require("./commands/CreateTrafficPolicyVersionCommand");
const CreateVPCAssociationAuthorizationCommand_1 = require("./commands/CreateVPCAssociationAuthorizationCommand");
const DeactivateKeySigningKeyCommand_1 = require("./commands/DeactivateKeySigningKeyCommand");
const DeleteCidrCollectionCommand_1 = require("./commands/DeleteCidrCollectionCommand");
const DeleteHealthCheckCommand_1 = require("./commands/DeleteHealthCheckCommand");
const DeleteHostedZoneCommand_1 = require("./commands/DeleteHostedZoneCommand");
const DeleteKeySigningKeyCommand_1 = require("./commands/DeleteKeySigningKeyCommand");
const DeleteQueryLoggingConfigCommand_1 = require("./commands/DeleteQueryLoggingConfigCommand");
const DeleteReusableDelegationSetCommand_1 = require("./commands/DeleteReusableDelegationSetCommand");
const DeleteTrafficPolicyCommand_1 = require("./commands/DeleteTrafficPolicyCommand");
const DeleteTrafficPolicyInstanceCommand_1 = require("./commands/DeleteTrafficPolicyInstanceCommand");
const DeleteVPCAssociationAuthorizationCommand_1 = require("./commands/DeleteVPCAssociationAuthorizationCommand");
const DisableHostedZoneDNSSECCommand_1 = require("./commands/DisableHostedZoneDNSSECCommand");
const DisassociateVPCFromHostedZoneCommand_1 = require("./commands/DisassociateVPCFromHostedZoneCommand");
const EnableHostedZoneDNSSECCommand_1 = require("./commands/EnableHostedZoneDNSSECCommand");
const GetAccountLimitCommand_1 = require("./commands/GetAccountLimitCommand");
const GetChangeCommand_1 = require("./commands/GetChangeCommand");
const GetCheckerIpRangesCommand_1 = require("./commands/GetCheckerIpRangesCommand");
const GetDNSSECCommand_1 = require("./commands/GetDNSSECCommand");
const GetGeoLocationCommand_1 = require("./commands/GetGeoLocationCommand");
const GetHealthCheckCommand_1 = require("./commands/GetHealthCheckCommand");
const GetHealthCheckCountCommand_1 = require("./commands/GetHealthCheckCountCommand");
const GetHealthCheckLastFailureReasonCommand_1 = require("./commands/GetHealthCheckLastFailureReasonCommand");
const GetHealthCheckStatusCommand_1 = require("./commands/GetHealthCheckStatusCommand");
const GetHostedZoneCommand_1 = require("./commands/GetHostedZoneCommand");
const GetHostedZoneCountCommand_1 = require("./commands/GetHostedZoneCountCommand");
const GetHostedZoneLimitCommand_1 = require("./commands/GetHostedZoneLimitCommand");
const GetQueryLoggingConfigCommand_1 = require("./commands/GetQueryLoggingConfigCommand");
const GetReusableDelegationSetCommand_1 = require("./commands/GetReusableDelegationSetCommand");
const GetReusableDelegationSetLimitCommand_1 = require("./commands/GetReusableDelegationSetLimitCommand");
const GetTrafficPolicyCommand_1 = require("./commands/GetTrafficPolicyCommand");
const GetTrafficPolicyInstanceCommand_1 = require("./commands/GetTrafficPolicyInstanceCommand");
const GetTrafficPolicyInstanceCountCommand_1 = require("./commands/GetTrafficPolicyInstanceCountCommand");
const ListCidrBlocksCommand_1 = require("./commands/ListCidrBlocksCommand");
const ListCidrCollectionsCommand_1 = require("./commands/ListCidrCollectionsCommand");
const ListCidrLocationsCommand_1 = require("./commands/ListCidrLocationsCommand");
const ListGeoLocationsCommand_1 = require("./commands/ListGeoLocationsCommand");
const ListHealthChecksCommand_1 = require("./commands/ListHealthChecksCommand");
const ListHostedZonesByNameCommand_1 = require("./commands/ListHostedZonesByNameCommand");
const ListHostedZonesByVPCCommand_1 = require("./commands/ListHostedZonesByVPCCommand");
const ListHostedZonesCommand_1 = require("./commands/ListHostedZonesCommand");
const ListQueryLoggingConfigsCommand_1 = require("./commands/ListQueryLoggingConfigsCommand");
const ListResourceRecordSetsCommand_1 = require("./commands/ListResourceRecordSetsCommand");
const ListReusableDelegationSetsCommand_1 = require("./commands/ListReusableDelegationSetsCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const ListTagsForResourcesCommand_1 = require("./commands/ListTagsForResourcesCommand");
const ListTrafficPoliciesCommand_1 = require("./commands/ListTrafficPoliciesCommand");
const ListTrafficPolicyInstancesByHostedZoneCommand_1 = require("./commands/ListTrafficPolicyInstancesByHostedZoneCommand");
const ListTrafficPolicyInstancesByPolicyCommand_1 = require("./commands/ListTrafficPolicyInstancesByPolicyCommand");
const ListTrafficPolicyInstancesCommand_1 = require("./commands/ListTrafficPolicyInstancesCommand");
const ListTrafficPolicyVersionsCommand_1 = require("./commands/ListTrafficPolicyVersionsCommand");
const ListVPCAssociationAuthorizationsCommand_1 = require("./commands/ListVPCAssociationAuthorizationsCommand");
const TestDNSAnswerCommand_1 = require("./commands/TestDNSAnswerCommand");
const UpdateHealthCheckCommand_1 = require("./commands/UpdateHealthCheckCommand");
const UpdateHostedZoneCommentCommand_1 = require("./commands/UpdateHostedZoneCommentCommand");
const UpdateTrafficPolicyCommentCommand_1 = require("./commands/UpdateTrafficPolicyCommentCommand");
const UpdateTrafficPolicyInstanceCommand_1 = require("./commands/UpdateTrafficPolicyInstanceCommand");
const Route53Client_1 = require("./Route53Client");
const commands = {
    ActivateKeySigningKeyCommand: ActivateKeySigningKeyCommand_1.ActivateKeySigningKeyCommand,
    AssociateVPCWithHostedZoneCommand: AssociateVPCWithHostedZoneCommand_1.AssociateVPCWithHostedZoneCommand,
    ChangeCidrCollectionCommand: ChangeCidrCollectionCommand_1.ChangeCidrCollectionCommand,
    ChangeResourceRecordSetsCommand: ChangeResourceRecordSetsCommand_1.ChangeResourceRecordSetsCommand,
    ChangeTagsForResourceCommand: ChangeTagsForResourceCommand_1.ChangeTagsForResourceCommand,
    CreateCidrCollectionCommand: CreateCidrCollectionCommand_1.CreateCidrCollectionCommand,
    CreateHealthCheckCommand: CreateHealthCheckCommand_1.CreateHealthCheckCommand,
    CreateHostedZoneCommand: CreateHostedZoneCommand_1.CreateHostedZoneCommand,
    CreateKeySigningKeyCommand: CreateKeySigningKeyCommand_1.CreateKeySigningKeyCommand,
    CreateQueryLoggingConfigCommand: CreateQueryLoggingConfigCommand_1.CreateQueryLoggingConfigCommand,
    CreateReusableDelegationSetCommand: CreateReusableDelegationSetCommand_1.CreateReusableDelegationSetCommand,
    CreateTrafficPolicyCommand: CreateTrafficPolicyCommand_1.CreateTrafficPolicyCommand,
    CreateTrafficPolicyInstanceCommand: CreateTrafficPolicyInstanceCommand_1.CreateTrafficPolicyInstanceCommand,
    CreateTrafficPolicyVersionCommand: CreateTrafficPolicyVersionCommand_1.CreateTrafficPolicyVersionCommand,
    CreateVPCAssociationAuthorizationCommand: CreateVPCAssociationAuthorizationCommand_1.CreateVPCAssociationAuthorizationCommand,
    DeactivateKeySigningKeyCommand: DeactivateKeySigningKeyCommand_1.DeactivateKeySigningKeyCommand,
    DeleteCidrCollectionCommand: DeleteCidrCollectionCommand_1.DeleteCidrCollectionCommand,
    DeleteHealthCheckCommand: DeleteHealthCheckCommand_1.DeleteHealthCheckCommand,
    DeleteHostedZoneCommand: DeleteHostedZoneCommand_1.DeleteHostedZoneCommand,
    DeleteKeySigningKeyCommand: DeleteKeySigningKeyCommand_1.DeleteKeySigningKeyCommand,
    DeleteQueryLoggingConfigCommand: DeleteQueryLoggingConfigCommand_1.DeleteQueryLoggingConfigCommand,
    DeleteReusableDelegationSetCommand: DeleteReusableDelegationSetCommand_1.DeleteReusableDelegationSetCommand,
    DeleteTrafficPolicyCommand: DeleteTrafficPolicyCommand_1.DeleteTrafficPolicyCommand,
    DeleteTrafficPolicyInstanceCommand: DeleteTrafficPolicyInstanceCommand_1.DeleteTrafficPolicyInstanceCommand,
    DeleteVPCAssociationAuthorizationCommand: DeleteVPCAssociationAuthorizationCommand_1.DeleteVPCAssociationAuthorizationCommand,
    DisableHostedZoneDNSSECCommand: DisableHostedZoneDNSSECCommand_1.DisableHostedZoneDNSSECCommand,
    DisassociateVPCFromHostedZoneCommand: DisassociateVPCFromHostedZoneCommand_1.DisassociateVPCFromHostedZoneCommand,
    EnableHostedZoneDNSSECCommand: EnableHostedZoneDNSSECCommand_1.EnableHostedZoneDNSSECCommand,
    GetAccountLimitCommand: GetAccountLimitCommand_1.GetAccountLimitCommand,
    GetChangeCommand: GetChangeCommand_1.GetChangeCommand,
    GetCheckerIpRangesCommand: GetCheckerIpRangesCommand_1.GetCheckerIpRangesCommand,
    GetDNSSECCommand: GetDNSSECCommand_1.GetDNSSECCommand,
    GetGeoLocationCommand: GetGeoLocationCommand_1.GetGeoLocationCommand,
    GetHealthCheckCommand: GetHealthCheckCommand_1.GetHealthCheckCommand,
    GetHealthCheckCountCommand: GetHealthCheckCountCommand_1.GetHealthCheckCountCommand,
    GetHealthCheckLastFailureReasonCommand: GetHealthCheckLastFailureReasonCommand_1.GetHealthCheckLastFailureReasonCommand,
    GetHealthCheckStatusCommand: GetHealthCheckStatusCommand_1.GetHealthCheckStatusCommand,
    GetHostedZoneCommand: GetHostedZoneCommand_1.GetHostedZoneCommand,
    GetHostedZoneCountCommand: GetHostedZoneCountCommand_1.GetHostedZoneCountCommand,
    GetHostedZoneLimitCommand: GetHostedZoneLimitCommand_1.GetHostedZoneLimitCommand,
    GetQueryLoggingConfigCommand: GetQueryLoggingConfigCommand_1.GetQueryLoggingConfigCommand,
    GetReusableDelegationSetCommand: GetReusableDelegationSetCommand_1.GetReusableDelegationSetCommand,
    GetReusableDelegationSetLimitCommand: GetReusableDelegationSetLimitCommand_1.GetReusableDelegationSetLimitCommand,
    GetTrafficPolicyCommand: GetTrafficPolicyCommand_1.GetTrafficPolicyCommand,
    GetTrafficPolicyInstanceCommand: GetTrafficPolicyInstanceCommand_1.GetTrafficPolicyInstanceCommand,
    GetTrafficPolicyInstanceCountCommand: GetTrafficPolicyInstanceCountCommand_1.GetTrafficPolicyInstanceCountCommand,
    ListCidrBlocksCommand: ListCidrBlocksCommand_1.ListCidrBlocksCommand,
    ListCidrCollectionsCommand: ListCidrCollectionsCommand_1.ListCidrCollectionsCommand,
    ListCidrLocationsCommand: ListCidrLocationsCommand_1.ListCidrLocationsCommand,
    ListGeoLocationsCommand: ListGeoLocationsCommand_1.ListGeoLocationsCommand,
    ListHealthChecksCommand: ListHealthChecksCommand_1.ListHealthChecksCommand,
    ListHostedZonesCommand: ListHostedZonesCommand_1.ListHostedZonesCommand,
    ListHostedZonesByNameCommand: ListHostedZonesByNameCommand_1.ListHostedZonesByNameCommand,
    ListHostedZonesByVPCCommand: ListHostedZonesByVPCCommand_1.ListHostedZonesByVPCCommand,
    ListQueryLoggingConfigsCommand: ListQueryLoggingConfigsCommand_1.ListQueryLoggingConfigsCommand,
    ListResourceRecordSetsCommand: ListResourceRecordSetsCommand_1.ListResourceRecordSetsCommand,
    ListReusableDelegationSetsCommand: ListReusableDelegationSetsCommand_1.ListReusableDelegationSetsCommand,
    ListTagsForResourceCommand: ListTagsForResourceCommand_1.ListTagsForResourceCommand,
    ListTagsForResourcesCommand: ListTagsForResourcesCommand_1.ListTagsForResourcesCommand,
    ListTrafficPoliciesCommand: ListTrafficPoliciesCommand_1.ListTrafficPoliciesCommand,
    ListTrafficPolicyInstancesCommand: ListTrafficPolicyInstancesCommand_1.ListTrafficPolicyInstancesCommand,
    ListTrafficPolicyInstancesByHostedZoneCommand: ListTrafficPolicyInstancesByHostedZoneCommand_1.ListTrafficPolicyInstancesByHostedZoneCommand,
    ListTrafficPolicyInstancesByPolicyCommand: ListTrafficPolicyInstancesByPolicyCommand_1.ListTrafficPolicyInstancesByPolicyCommand,
    ListTrafficPolicyVersionsCommand: ListTrafficPolicyVersionsCommand_1.ListTrafficPolicyVersionsCommand,
    ListVPCAssociationAuthorizationsCommand: ListVPCAssociationAuthorizationsCommand_1.ListVPCAssociationAuthorizationsCommand,
    TestDNSAnswerCommand: TestDNSAnswerCommand_1.TestDNSAnswerCommand,
    UpdateHealthCheckCommand: UpdateHealthCheckCommand_1.UpdateHealthCheckCommand,
    UpdateHostedZoneCommentCommand: UpdateHostedZoneCommentCommand_1.UpdateHostedZoneCommentCommand,
    UpdateTrafficPolicyCommentCommand: UpdateTrafficPolicyCommentCommand_1.UpdateTrafficPolicyCommentCommand,
    UpdateTrafficPolicyInstanceCommand: UpdateTrafficPolicyInstanceCommand_1.UpdateTrafficPolicyInstanceCommand,
};
class Route53 extends Route53Client_1.Route53Client {
}
exports.Route53 = Route53;
(0, smithy_client_1.createAggregatedClient)(commands, Route53);
