// @pulumi/cloudflare/config/vars.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const __config = new pulumi.Config("cloudflare");
Object.defineProperty(exports, "apiKey", {
    get() {
        return __config.get("apiKey");
    },
    enumerable: true,
});
Object.defineProperty(exports, "apiToken", {
    get() {
        return __config.get("apiToken");
    },
    enumerable: true,
});
Object.defineProperty(exports, "apiUserServiceKey", {
    get() {
        return __config.get("apiUserServiceKey");
    },
    enumerable: true,
});
Object.defineProperty(exports, "baseUrl", {
    get() {
        return __config.get("baseUrl");
    },
    enumerable: true,
});
Object.defineProperty(exports, "email", {
    get() {
        return __config.get("email");
    },
    enumerable: true,
});
Object.defineProperty(exports, "userAgentOperatorSuffix", {
    get() {
        return __config.get("userAgentOperatorSuffix");
    },
    enumerable: true,
});
