// @pulumiverse/vercel/webhook.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhook = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * A webhook is a trigger-based HTTP endpoint configured to receive HTTP POST requests through events.
 *
 * When an event happens, a webhook is sent to a third-party app, which can then take appropriate action.
 *
 * > Only Pro and Enterprise teams are able to configure these webhooks at the account level.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as vercel from "@pulumiverse/vercel";
 *
 * const example = new vercel.Project("example", {});
 * const example2 = new vercel.Project("example2", {});
 * const withProjectIds = new vercel.Webhook("withProjectIds", {
 *     events: [
 *         "deployment.created",
 *         "deployment.succeeded",
 *     ],
 *     endpoint: "https://example.com/endpoint",
 *     projectIds: [
 *         example.id,
 *         example2.id,
 *     ],
 * });
 * const withoutProjectIds = new vercel.Webhook("withoutProjectIds", {
 *     events: [
 *         "deployment.created",
 *         "deployment.succeeded",
 *     ],
 *     endpoint: "https://example.com/endpoint",
 * });
 * ```
 */
class Webhook extends pulumi.CustomResource {
    /**
     * Get an existing Webhook resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Webhook(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Webhook.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Webhook.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["endpoint"] = state ? state.endpoint : undefined;
            resourceInputs["events"] = state ? state.events : undefined;
            resourceInputs["projectIds"] = state ? state.projectIds : undefined;
            resourceInputs["secret"] = state ? state.secret : undefined;
            resourceInputs["teamId"] = state ? state.teamId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.endpoint === undefined) && !opts.urn) {
                throw new Error("Missing required property 'endpoint'");
            }
            if ((!args || args.events === undefined) && !opts.urn) {
                throw new Error("Missing required property 'events'");
            }
            resourceInputs["endpoint"] = args ? args.endpoint : undefined;
            resourceInputs["events"] = args ? args.events : undefined;
            resourceInputs["projectIds"] = args ? args.projectIds : undefined;
            resourceInputs["teamId"] = args ? args.teamId : undefined;
            resourceInputs["secret"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["secret"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(Webhook.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Webhook = Webhook;
/** @internal */
Webhook.__pulumiType = 'vercel:index/webhook:Webhook';
